packages/cli/src/databases/repositories/execution.repository.ts

```diff
@@ -487,7 +503,13 @@ export class ExecutionRepository extends Repository<ExecutionEntity> {
 		await this.createQueryBuilder()
 			.update(ExecutionEntity)
 			.set({ deletedAt: new Date() })
-			.where(
+			.where({
+				// Don't re-mark already deleted executions
+				deletedAt: IsNull(),
+				// Only mark executions as deleted if they are in an end state
+				status: In(['canceled', 'crashed', 'error', 'failed', 'success']),
```

Comment:
What about `unknown` and `warning`? Should we maybe flip this and prune anything that is not `new`, `running` or `waiting`?

Reply:
You tell me. I have no idea when executions end up in those states and if they are end-states. I think pruning everything other than `new`, `running` and `waiting` would probably make sense. At least it's an improvement to the current state. I'll change that